commit 6ce43f2a6fa2b7afc12a8d8d72313305ff468356 (refs/changes/46/2446/3)
Author: Jerry Jelinek <jerry.jelinek@joyent.com>
Date:   2017-08-23T19:44:13+00:00 (2 years, 2 months ago)
    
    OS-6275 lx pipe buffer is too small
    Reviewed by: Patrick Mooney <patrick.mooney@joyent.com>
    Approved by: Patrick Mooney <patrick.mooney@joyent.com>

diff --git a/usr/src/uts/common/brand/lx/syscall/lx_pipe.c b/usr/src/uts/common/brand/lx/syscall/lx_pipe.c
index d6c8f1d274..cdada58f15 100644
--- a/usr/src/uts/common/brand/lx/syscall/lx_pipe.c
+++ b/usr/src/uts/common/brand/lx/syscall/lx_pipe.c
@@ -179,6 +179,15 @@ lx_hd_pipe(intptr_t arg, int flags)
 	VERIFY((str = vp1->v_stream) != NULL);
 	(void) lx_pipe_setsz(str, LX_DEFAULT_PIPE_SIZE, B_TRUE);
 
+	/*
+	 * Because we're using streams to increase the capacity of the pipe
+	 * up to the default Linux capacity, we have to switch the pipe
+	 * out of FIFOFAST mode and back to a normal stream. In fast mode
+	 * the pipe capacity is limited to "Fifohiwat" which is a compile-time
+	 * limit set to FIFOHIWAT.
+	 */
+	fifo_fastoff(VTOF(vp1));
+
 	/*
 	 * Set the O_NONBLOCK flag if requested.
 	 */
